### PR TITLE
Replaces go-autorest MS graph client with msgraph-sdk-go

### PR DESCRIPTION
**Pull request by austingebauer**
## Overview

This PR replaces the deprecated [go-autorest](https://github.com/Azure/go-autorest) module with [msgraph-sdk-go](https://github.com/microsoftgraph/msgraph-sdk-go). Additionally, it migrates the [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/MIGRATION_GUIDE.md) to the new package structure (see related [issue](https://github.com/hashicorp/vault-plugin-secrets-azure/issues/143)).

There are no known API changes or compatibility breaks in this PR.

## Testing

All tests in the repository are passing. Additionally, I manually tested the secrets engine functionality using our Terraform [bootstrap](https://github.com/hashicorp/vault-plugin-auth-azure/tree/main/bootstrap/terraform) which pulls an [IMDS token](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token) off of a VM.

I've additionally added an acceptance test that actually uses Azure resources to login in.

## Related Issues/Pull Requests
- https://github.com/hashicorp/vault-plugin-secrets-azure/issues/143

## Contributor Checklist

- [x] Add output for any tests not ran in CI to the PR description (eg, acceptance tests)
- [x] Backwards compatible

### CODE REVIEW

File: azure.go [hunk 1]
```diff
@@ -32,38 +29,21 @@ import (
...
-}
-
-type providersClient interface {
-	Get(ctx context.Context, resourceProviderNamespace string, options *armresources.ProvidersClientGetOptions) (armresources.ProvidersClientGetResponse, error)
-}
-
-type tokenVerifier interface {
-	Verify(ctx context.Context, token string) (*oidc.IDToken, error)
-}
+// https://learn.microsoft.com/en-us/graph/sdks/national-clouds
```
Comment (austingebauer) [inline azure.go:32]: Couldn't find constants for these in the library code.

File: azure.go [hunk 2]
```diff
@@ -391,3 +361,19 @@ func ConfigurationFromName(name string) (cloud.Configuration, error) {
 
 	return c, nil
 }
+
+func graphURIFromName(name string) (string, error) {
+	configs := map[string]string{
+		"AZURECHINACLOUD":        azureChinaCloudBaseURI,
```
Comment (raymonstah) [inline azure.go:367]: it might be worthwhile to make these constants as well

**Review by raymonstah — APPROVED** [review]
Looks good! question about the error handling: I noticed that a lot of the `if err` checks return the error directly. Do we feel that the error returned provides enough context? If not, we can consider wrapping them.

File: azure.go (hunk 2 above)
Comment (austingebauer) [reply azure.go]: Good idea. Done in https://github.com/hashicorp/vault-plugin-auth-azure/pull/121/commits/2b2bf18974cc94962f388ff8a574c9579925dd90.

**Comment by austingebauer** [comment]
Good point @raymonstah. Checking some locations to see if we can provide more error context.

**Review by vinay-gopalan — APPROVED** [review]
Looks great, thanks for working through this!

**Pull request merged**